byzcoin/service.go

```diff
@@ -35,9 +35,19 @@ import (
 // not trigger, and the acceptable window would be ± 30 sec.
 var minTimestampWindow = 10 * time.Second
 
+// For tests to influence when the whole collection will be downloaded if
+// some blocks are missing.
+var catchupDownloadAll = 100
```

Comment:
Those parameters shouldn't be in the service struct ? I understand you want to have the "constants" at the beginning but when a different service wants to test and change those accordingly, it can't because it's hidden.
You could have the default value as a constant at the top, that with a setter for what can be changed.

Reply:
these values only have to be changed by byzcoin-tests, not other tests. If we need to change them for other tests, too, then what you propose makes sense, e.g., `SetPropagationTimeout`